src/components/Pagination/Pagination.vue

```diff
@@ -0,0 +1,315 @@
+<template>
+  <div
+    v-if="shouldRender"
+    class="pagination"
+  >
+    <div>
+      <p class="pagination__results">
+        {{ totalNumber }} result<span v-if="totalNumber !== 1">s</span>
+      </p>
+    </div>
+
+    <div class="pagination__paging-container">
+      <p class="pagination__text">
+        {{ paginationText }}
+      </p>
+      <button
+        :class="[
+          'pagination__btn',
+          { 'pagination__btn--disabled': page === 1 },
+        ]"
+        :disabled="page === 1"
+        @click="pageClick(1)"
+      >
+        <span class="sr-only">Go to first page</span>
+        <svg
+          width="16"
+          height="16"
+          viewBox="0 0 16 16"
+          fill="none"
+          xmlns="http://www.w3.org/2000/svg"
+        >
+          <path
+            fill-rule="evenodd"
+            clip-rule="evenodd"
+            d="M12.6451 14.0024L6.86243 8.36551C6.69482 8.20053 6.69482 7.92556 6.86243 7.76058L12.6451 2.12374C12.8127 1.95875 13.0921 1.95875 13.2597 2.12374L13.8743 2.72867C14.0419 2.89365 14.0419 3.16861 13.8743 3.3336L9.32077 7.76058C9.15316 7.92556 9.15316 8.20053 9.32077 8.36551L13.8743 12.7925C14.0419 12.9575 14.0419 13.2324 13.8743 13.3974L13.2597 14.0024C13.0642 14.1398 12.8127 14.1398 12.6451 14.0024Z"
+          />
+          <path
+            fill-rule="evenodd"
+            clip-rule="evenodd"
+            d="M3.96543 2C4.15873 2 4.31543 2.1567 4.31543 2.35L4.31543 13.7555C4.31543 13.9488 4.15873 14.1055 3.96543 14.1055L2.84961 14.1055C2.65631 14.1055 2.49961 13.9488 2.49961 13.7555L2.49961 2.35C2.49961 2.1567 2.65631 2 2.84961 2L3.96543 2Z"
+          />
+        </svg>
+      </button>
+      <button
+        :class="['pagination__btn', { 'pagination__btn--disabled': page <= 1 }]"
```

Comment:
Nitpick: this seems to be the only class in this file where everything's on the same line. I'll see if I can configure something in the linter to catch this automatically... 🤔